internal/controller/clusterdeployment_reconciler_test.go

```diff
@@ -0,0 +1,370 @@
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(c.Get(ctx, key, siteConfig)).To(Succeed())
+		conditions.SetStatusCondition(&siteConfig.Status.Conditions,
+			conditions.Provisioned,
+			conditions.InProgress,
+			metav1.ConditionTrue,
+			"Provisioning cluster")
+		err = conditions.UpdateStatus(ctx, c, siteConfig)
+		Expect(err).ToNot(HaveOccurred())
+
+		sc := &v1alpha1.SiteConfig{}
+		Expect(c.Get(ctx, key, sc)).To(Succeed())
+
+		DeploymentConditions := [][]hivev1.ClusterDeploymentCondition{
+			{
+				{
+					Type:    hivev1.ClusterInstallRequirementsMetClusterDeploymentCondition,
+					Status:  corev1.ConditionStatus(metav1.ConditionFalse),
+					Reason:  "ClusterNotReady",
+					Message: "The cluster is not ready to begin the installation",
+				},
+				{
+					Type:    hivev1.ClusterInstallStoppedClusterDeploymentCondition,
+					Status:  corev1.ConditionStatus(metav1.ConditionFalse),
+					Reason:  "InstallationNotStopped",
+					Message: "The installation is waiting to start or in progress",
+				},
+				{
+					Type:    hivev1.ClusterInstallCompletedClusterDeploymentCondition,
+					Status:  corev1.ConditionStatus(metav1.ConditionTrue),
+					Reason:  "InstallationNotFailed",
+					Message: "The installation has not started",
+				},
+				{
+					Type:    hivev1.ClusterInstallFailedClusterDeploymentCondition,
+					Status:  corev1.ConditionStatus(metav1.ConditionFalse),
+					Reason:  "InstallationNotFailed",
+					Message: "The installation has not started",
+				},
+			},
+
+			{
+				{
+					Type:    hivev1.ClusterInstallRequirementsMetClusterDeploymentCondition,
+					Status:  corev1.ConditionStatus(metav1.ConditionTrue),
+					Reason:  "ClusterInstallationStopped",
+					Message: "The cluster installation stopped",
+				},
+				{
+					Type:    hivev1.ClusterInstallStoppedClusterDeploymentCondition,
+					Status:  corev1.ConditionStatus(metav1.ConditionTrue),
+					Reason:  "ClusterInstallStopped",
+					Message: "The installation has stopped because it completed successfully",
+				},
+				{
+					Type:    hivev1.ClusterInstallCompletedClusterDeploymentCondition,
+					Status:  corev1.ConditionStatus(metav1.ConditionTrue),
+					Reason:  "InstallationCompleted",
+					Message: "The installation has completed: Cluster is installed",
+				},
+				{
+					Type:    hivev1.ClusterInstallFailedClusterDeploymentCondition,
+					Status:  corev1.ConditionStatus(metav1.ConditionFalse),
+					Reason:  "InstallationNotFailed",
+					Message: "The installation has not failed",
+				},
+			},
+		}
+
+		for _, deploymentCondition := range DeploymentConditions {
+			clusterDeployment.Status.Conditions = deploymentCondition
+			Expect(c.Update(ctx, clusterDeployment)).To(Succeed())
+
+			res, err := r.Reconcile(ctx, ctrl.Request{NamespacedName: key})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(res).To(Equal(requeueImmediately()))
+
+			sc := &v1alpha1.SiteConfig{}
+			Expect(c.Get(ctx, key, sc)).To(Succeed())
+
+			matchedConditions := true
```

Comment:
Same as the previous comment. Assert on the condition data in the type conditional and annotate the expectations that are just bools.